aviatrix/resource_aviatrix_transit_gateway_peering.go

```diff
@@ -124,6 +134,12 @@ func resourceAviatrixTransitGatewayPeeringCreate(d *schema.ResourceData, meta in
 		PrivateIPPeering:               d.Get("enable_peering_over_private_network").(bool),
 	}
 
+	if transitGatewayPeering.PrivateIPPeering {
+		transitGatewayPeering.SingleTunnel = d.Get("enable_single_tunnel_mode").(bool)
+	} else {
+		return fmt.Errorf("enable_single_tunnel_mode is only valid when enable_peering_over_private_network is set to true")
+	}
+
```

Comment:
Whenever `enable_peering_over_private_network` is false we return an error?

Reply:
When `enable_peering_over_private_network` is false, because of `DiffSuppressFunc`, `enable_single_tunnel_mode` will be ignored. Do we still need to return an error?

Reply:
Have you tested creating a peering with enable_peering_over_private_network set to False? The code looks like whenever enable_peering_over_private_network is false you return an error

Reply:
Also this is in the Create function so the diffsuppressfunc does not mean anything

Reply:
The logic here is wrong for sure, this will be fixed. 
When I try terraform plan, if the private_network is false, it seems the single_tunnel attribute is ignored directly. So the diffsuppressfunc works in Create?

`shuoliu@Shuos-MBP matis_0012272 % terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.


------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aviatrix_transit_gateway_peering.foo will be created
  + resource "aviatrix_transit_gateway_peering" "foo" {
      + enable_peering_over_private_network = false
      + id                                  = (known after apply)
      + transit_gateway_name1               = "test1"
      + transit_gateway_name2               = "test2"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------`